Antidote/CallControlsView.m

```diff
@@ -91,28 +113,44 @@ - (void)installConstraints
     [self.resumeButton makeConstraints:^(MASConstraintMaker *make) {
         make.right.equalTo(self.videoButton.left).with.offset(-k3ButtonGap);
         make.centerY.equalTo(self.videoButton);
-        make.size.equalTo(kButtonSide);
+        make.size.equalTo(self.videoButton);
     }];
 
     [self.microphoneButton makeConstraints:^(MASConstraintMaker *make) {
-        make.left.equalTo(self);
+        make.left.equalTo(self.expandedView);
         make.right.equalTo(self.speakerButton.left).with.offset(-k3ButtonGap);
-        make.size.equalTo(kButtonSide);
-        make.top.equalTo(self.videoButton.bottom).with.offset(k3ButtonGap);
+        make.size.equalTo(self.videoButton);
+        make.bottom.equalTo(self.expandedView);
     }];
 
     [self.speakerButton makeConstraints:^(MASConstraintMaker *make) {
-        make.right.equalTo(self);
-        make.size.equalTo(kButtonSide);
+        make.right.equalTo(self.expandedView);
+        make.size.equalTo(self.videoButton);
         make.centerY.equalTo(self.microphoneButton);
+        make.bottom.equalTo(self.expandedView);
     }];
 
-    [self makeConstraints:^(MASConstraintMaker *make) {
-        make.top.equalTo(self.videoButton);
-        make.bottom.equalTo(self.microphoneButton);
+    [self.expandedView makeConstraints:^(MASConstraintMaker *make) {
+        make.centerY.equalTo(self);
+        make.height.equalTo(kExpandedViewHeight);
+        make.left.equalTo(self).with.offset(kExpandedViewWidthIndents);
+        make.right.equalTo(self).with.offset(-kExpandedViewWidthIndents);
+    }];
+
+    [self.compactView makeConstraints:^(MASConstraintMaker *make) {
+        make.bottom.equalTo(self);
+        make.left.equalTo(self);
+        make.right.equalTo(self);
     }];
 }
 
+- (void)layoutSubviews
+{
+    [super layoutSubviews];
+    [self layoutIfNeeded];
```

Comment:
Subviews are already layout-ed here. Why do you need to call `layoutIfNeeded` again?